src/dlsproto/node/neo/request/GetRange.d

```diff
@@ -12,13 +12,527 @@
+    {
+        this.connection = connection;
+        this.ed = connection.event_dispatcher;
+        this.parser = this.ed.message_parser;
+
+        cstring channel_name;
+        cstring filter_string;
+        time_t low, high;
+        Filter.FilterMode filter_mode;
+
+        this.parser.parseBody(msg_payload, channel_name, low, high,
+                filter_string, filter_mode);
+
+        if ( !this.prepareChannel(channel_name) )
+        {
+            this.ed.send(
+                ( ed.Payload payload )
+                {
+                    payload.addConstant(RequestStatusCode.Error);
+                }
+            );
+            return;
+        }
+
+        if ( !this.prepareRange(low, high) )
+        {
+            this.ed.send(
+                ( ed.Payload payload )
+                {
+                    payload.addConstant(RequestStatusCode.Error);
+                }
+            );
+            return;
+        }
+
+        if (filter_string.length > 0)
+        {
+            if ( !this.prepareFilter(filter_mode, filter_string) )
+            {
+                this.ed.send(
+                    ( ed.Payload payload )
+                    {
+                        payload.addConstant(RequestStatusCode.Error);
+                    }
+                );
+                return;
+            }
+        }
+
+        this.ed.send(
+            ( ed.Payload payload )
+            {
+                payload.addConstant(RequestStatusCode.Started);
+            }
+        );
+
+        this.value_buffer = this.resources.getVoidBuffer();
+        this.batch_buffer = this.resources.getVoidBuffer();
+        this.compressed_batch = this.resources.getVoidBuffer();
+        this.saved_exception = this.resources.getException();
+        this.lzo = this.resources.getLzo();
+
+        try
+        {
+            this.run();
+        }
+        finally
+        {
+            this.requestFinished();
+        }
+    }
+
+    /***************************************************************************
+
+        Request main loop. Gets records from the storage, sends them to the
+        client in batches and handles Continue/Stop client feedback.
+
+    ***************************************************************************/
+
+    private void run ()
+    {
+    main_loop: while (true)
+        {
+            time_t record_timestamp;
+
+            bool wait_for_data;
+            auto got_next = this.getNextRecord(record_timestamp, *this.value_buffer,
+                    wait_for_data);
+
+            if (wait_for_data)
+            {
+                auto event = this.ed.nextEvent(ed.NextEventFlags.Receive);
+
+                switch (event.active)
+                {
+                    // Resumed by the storage engine, now ready to return value
+                    // with `getNextRecord`, so we try again.
+                    case event.active.resumed:
+                        continue main_loop;
```

Comment:
The `main_loop` label is not needed.

Reply:
Yeah, it's _continue_ not _break_ :face_with_head_bandage:

Reply:
Updated.